com.unity.perception/Runtime/Randomization/Samplers/ISampler.cs

```diff
@@ -28,5 +28,15 @@ public interface ISampler
         /// <param name="jobHandle">The handle of the scheduled job</param>
         /// <returns>A NativeArray of generated samples</returns>
         NativeArray<float> Samples(int sampleCount, out JobHandle jobHandle);
+
+        /// <summary>
+        /// Used for performing sampler specific clean-up tasks (e.g. once the scenario is complete).
+        /// </summary>
+        void Cleanup();
+
+        /// <summary>
+        /// Used for performing sampler specific initialization tasks.
+        /// </summary>
+        void Initialize();
```

Comment:
While I'm all for optimization, I think enforcing an assumed lifecycle on samplers will get us into trouble. For example,  the split grammar uses samplers to generate random values in the editor to visualize intermediate results. The animation curve sampler would uninitialized in this scenario, since there is no game loop in the editor to call these lifecycle functions.

So I'd suggest the following:
- Check if the sampler is initialized in the sample function itself and cache any information there instead of calling the Initialize() method on all samplers from the scenario when the game loop begins
- The cleanup step as well can be replaced with a finalizer that disposes of the native array when the object is destroyed.
- Remove Initialize() and Cleanup() from the sampler interface

Example finalizer:
```
public class AnimationCurveSampler : ISampler
{
  public NativeArray<float> cachedData;

  ~AnimationCurveSampler()  // finalizer
  {
      cachedData.Dispose();
  }
}
```